src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperRunner.cs

```diff
@@ -24,7 +24,10 @@ public class TagHelperRunner
                 executionContext.AllAttributes,
                 executionContext.UniqueId,
                 executionContext.GetChildContentAsync);
-            var tagHelperOutput = new TagHelperOutput(executionContext.TagName, executionContext.HTMLAttributes);
+            var tagHelperOutput = new TagHelperOutput(executionContext.TagName, executionContext.HTMLAttributes)
+            {
+                SelfClosing = executionContext.SelfClosing,
```

Comment:
@NTaylorMullen there are no parallels to `TagBuilder` here because self-closing has no "original" source in the `TagBuilder` case -- it's completely up to the calling code.  in the `TagHelperOutput` case, self-closing comes from one and only one place in 90% of the cases.  yes `TagHelperOutput.SelfClosing` can be overridden (primarily for unusual wrapping scenarios) but there's a clear original value provided in the Razor source file.